### ISSUE TITLE
Autocorrect introduces violation - possible bug in Style/SingleLineMethods

### ISSUE THREAD

**Issue by wendelscardua** [bbatsov/rubocop#5853]
When autocorrecting the snippet below, rubocop introduces a new violation, `Layout/DefEndAlignment`

Original snippet:
```ruby
x.instance_eval { def foo; 413; end }
```

## Expected behavior

```ruby
x.instance_eval do
  def foo
    413
  end
end
```
## Actual behavior

```ruby
x.instance_eval do
  def foo
    413
                  end
end
```
## Steps to reproduce the problem

1. Save the snippet to `example.rb`
2. Run `rubocop -a example.rb`

## RuboCop version

```
$ rubocop -V
0.55.0 (using Parser 2.5.1.0, running on ruby 2.3.5 x86_64-linux)
```

**Comment by wendelscardua** [comment]
PS: the same happens to the snippet below:
```ruby
def foo; def bar; 413; end; end
```
(although this one is less likely to happen in real code)

**Comment by Drenmi** [comment]
When RuboCop makes an auto-correction, it runs all the other cops again on the corrected file, to pick up new offenses. The reasons this isn't being corrected is because `Lint/EndAlignment` has auto-correct disabled by default. From the conversation:

> The end alignment cops are in the Lint category because the bad indentation could be something more serious than just a style issue. It could be a mistake in which keyword you think you're matching with the end. (ruby -w warns for these too, by the way.) So for this reason I don't think we can add auto-correct for these cops.
>
> Maybe if auto-correct could be disabled by default in configuration.

If the auto-correction is enabled, this is fixed accordingly. You may or may not want to enable it based on the above comment.
